tests/devices/qubit/test_sampling.py

```diff
@@ -490,6 +533,145 @@ def test_measure_with_samples_one_shot_one_wire(self):
         assert result == -1.0
 
 
+class TestInvalidStateSamples:
+    """Tests for state vectors containing nan values or shot vectors with zero shots."""
+
+    @pytest.mark.parametrize("shots", [10, [10, 10]])
+    def test_only_catch_nan_errors(self, shots):
+        """Test that errors are only caught if they are raised due to nan values in the state."""
+        state = np.zeros((2, 2)).astype(np.complex128)
+        mp = qml.expval(qml.PauliZ(0))
+        _shots = Shots(shots)
+
+        with pytest.raises(Exception):
```

Comment:
can you catch something more specific here?

Reply:
The point of the test is to check that ALL errors except for the `probabilities contain NaN` error are caught, so it makes no difference.

Reply:
yeah but.. you're only catching one error here, right?

Reply:
Valid point, I'll update it.